src/nile/nre.py

```diff
@@ -33,6 +33,12 @@ def call(self, address_or_alias, method, params=None, abi=None):
             address_or_alias, "call", method, params, self.network, abi=abi
         )
 
+    async def execute_tx(self, tx_wrapper, watch_mode=None):
```

Comment:
Keeping in mind that this method is modifying the tx (not only executing it), I would rename it to something more descriptive. Not sure what name to use though, maybe `set_estimated_fee_and_execute`? This name doesn't take into account that if the fee is different than 0 the estimated one won't be used, so I think is still confusing.

Keeping in mind that execute_tx updates `tx.hash`, `tx.max_fee`, and `tx.query_hash`, users can get unexpected behavior on the script by calling this `execute_tx` method without knowing that it modifies the tx object.

I'm more in favor of keeping only this way, which is more explicit and easier to follow because of that:
```
tx = await account.send(ETH_TOKEN_ADDRESS, "transfer", [recipient, *amount])
max_fee = await tx.estimate_fee()
await tx.update_fee(max_fee).execute()
```

Reply:
> Keeping in mind that this method is modifying the tx (not only executing it), I would rename it to something more descriptive

Agreed. The idea was to have a simple and easy-to-use method. It's probably better to have `set_estimated_fee_and_execute` if we keep this approach

> users can get unexpected behavior on the script by calling this execute_tx method without knowing that it modifies the tx object

> I'm more in favor of keeping only this way, which is more explicit and easier to follow because of that:

My only contention with this consists of the "auto calculate fee" issue itself because we would be asking the user to set up the `update_fee` and it wouldn't exactly be automated in the NRE (though, the flow makes it quite easy). I suppose the question is: do you think the `tx.update_fee` flow is enough to satisfy the issue?

Reply:
Got you. I think having the current design, in NRE `update_fee` is enough to satisfy because issues by misleading tx hashes after execution weigh more than the benefit over the two extra lines (in CLI happens transparently).

Maybe we could implement an "auto fee" mechanism in Transaction initialization that would initialize the transaction with the estimated fee, transparently to the user (if the max_fee in construction time is set to -1 or "auto" in initialization for example), what do you think?

Reply:
I think that's a better idea! And it's implemented, I just need to update tests